checker/src/main/java/org/checkerframework/checker/calledmethods/builder/AutoValueSupport.java

```diff
@@ -0,0 +1,458 @@
+            possiblePropNames = new String[] {prop, Introspector.decapitalize(prop.substring(3))};
+        } else if (prop.startsWith("is")
+                && prop.length() > 2
+                && Character.isUpperCase(prop.charAt(2))) {
+            possiblePropNames = new String[] {prop, Introspector.decapitalize(prop.substring(2))};
+        } else {
+            possiblePropNames = new String[] {prop};
+        }
+
+        for (String propName : possiblePropNames) {
+            // in each case, the setter may be the property name itself, or prefixed by 'set'
+            String[] setterNamesToTry =
+                    new String[] {
+                        propName, "set" + BuilderFrameworkSupportUtils.capitalize(propName)
+                    };
+            for (String setterName : setterNamesToTry) {
+                if (builderSetterNames.contains(setterName)) {
+                    return setterName;
+                }
+            }
+        }
+
+        // Could not find a corresponding setter.  This is likely because an AutoValue Extension is
+        // in use.  See https://github.com/kelloggm/object-construction-checker/issues/110 .
+        // For now we return null, but once that bug is fixed, this should be changed to an
+        // assertion failure.
+        return null;
+    }
+
+    /**
+     * Computes the required properties of an @AutoValue class.
+     *
+     * @param autoValueClassElement the @AutoValue class
+     * @param avBuilderSetterNames names of all setters in the corresponding AutoValue builder class
+     * @return a list of required property names
+     */
+    private List<String> getAutoValueRequiredProperties(
+            final TypeElement autoValueClassElement, Set<String> avBuilderSetterNames) {
+        return getAllAbstractMethods(autoValueClassElement).stream()
+                .filter(member -> isAutoValueRequiredProperty(member, avBuilderSetterNames))
+                .map(e -> e.getSimpleName().toString())
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Does member represent a required property of an AutoValue class?
+     *
+     * @param member a member of an AutoValue class or superclass
+     * @param avBuilderSetterNames names of all setters in corresponding AutoValue builder class
+     * @return true if {@code member} is required
+     */
+    private boolean isAutoValueRequiredProperty(Element member, Set<String> avBuilderSetterNames) {
+        String name = member.getSimpleName().toString();
+        // Ignore java.lang.Object overrides, constructors, and toBuilder methods in AutoValue
+        // classes.
+        // Strictly speaking, this code should check return types, etc. to handle strange
+        // overloads and other corner cases. They seem unlikely enough that we are skipping for now.
+        if (ArraysPlume.indexOf(
+                        new String[] {"equals", "hashCode", "toString", "<init>", "toBuilder"},
```

Comment:
In your previous review (here: https://github.com/typetools/checker-framework/pull/3529#discussion_r476477088), you suggested inlining the existing final static field because it is only used once, which directly contradicts this suggestion.

Reply:
Oops!  Sorry about that.  I must not have been thinking about the cost of recomputing it.  I don't have a feel for whether the run-time cost is actually a problem -- I just worried about the extra allocation and garbage collection.

Reply:
I think the benefit to readability is worth the little bit of extra allocation/deallocation. This method is only called when actually analyzing a class that AutoValue generated (which shouldn't be that many classes), so the runtime overhead shouldn't be too bad.